python/hsfs/core/arrow_flight_client.py

```diff
@@ -191,52 +328,81 @@ def afs_error_handler_wrapper(instance, *args, **kw):
                     return func(instance, *args, **kw)
                 except Exception as e:
                     message = str(e)
+                    _logger.debug("Caught exception in %s: %s", func.__name__, message)
+                    _logger.exception(e)
                     if (
                         isinstance(e, FlightServerError)
                         and "Please register client certificates first." in message
                     ):
                         instance._register_certificates()
                         return func(instance, *args, **kw)
+                    elif (
+                        isinstance(e, FlightServerError)
+                        and "no free slot available for" in message
+                    ):
+                        raise FeatureStoreException(
+                            "No free slots available in Hopsworks Feature Query Service. Please try again later."
```

Comment:
imo `Hopsworks Feature Query Service is busy right now` is easier to understand.

Reply:
Ok